ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,898 @@
+            let mut x_start = 0;
+            let mut y_start = 0;
+            let mut x_end = 0;
+            let mut y_end = 0;
+            let r = ctru_sys::CAMU_GetTrimmingParams(
+                &mut x_start,
+                &mut y_start,
+                &mut x_end,
+                &mut y_end,
+                self.port_as_raw(),
+            );
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(CamTrimmingParams {
+                    x_start,
+                    y_start,
+                    x_end,
+                    y_end,
+                })
+            }
+        }
+    }
+
+    fn set_trimming_params_center(
+        &self,
+        trim_width: i16,
+        trim_height: i16,
+        cam_width: i16,
+        cam_height: i16,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetTrimmingParamsCenter(
+                self.port_as_raw(),
+                trim_width,
+                trim_height,
+                cam_width,
+                cam_height,
+            );
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn set_exposure(&mut self, exposure: i8) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetExposure(self.camera_as_raw(), exposure);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn set_white_balance(&mut self, white_balance: CamWhiteBalance) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetWhiteBalance(self.camera_as_raw(), white_balance.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn set_white_balance_without_base_up(
+        &mut self,
+        white_balance: CamWhiteBalance,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetWhiteBalanceWithoutBaseUp(
+                self.camera_as_raw(),
+                white_balance.bits(),
+            );
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn set_sharpness(&mut self, sharpness: i8) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetSharpness(self.camera_as_raw(), sharpness);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn set_auto_exposure(&mut self, enabled: bool) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetAutoExposure(self.camera_as_raw(), enabled);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    fn is_auto_exposure(&self) -> crate::Result<bool> {
```

Comment:
nit: Maybe rename to `is_auto_exposure_enabled`?
Similar functions have the same issue.